apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -1158,6 +1286,339 @@ impl FetchDependencyGraph {
+            && node_inputs.contains(grand_child_inputs))
+    }
+
+    /// Merges a child of parent node into it.
+    ///
+    /// Note that it is up to the caller to know that doing such merging is reasonable in the first place, which
+    /// generally means knowing that 1) `child.inputs` are included in `parent.inputs` and 2) all of `child.selection`
+    /// can safely be queried on the `parent.subgraphName` subgraph.
+    ///
+    /// Arguments:
+    /// * parent_id - parent node ID
+    /// * child_id - a node that must be a `child` of this parent, and for which the 'path in parent' (for given parent) is
+    ///   known. The `can_merge_child_in` method can be used to ensure that `child` meets those requirement.
+    fn merge_child_in(
+        &mut self,
+        node_id: NodeIndex,
+        child_id: NodeIndex,
+    ) -> Result<(), FederationError> {
+        let Some(relation_to_child) = self.parent_relation(child_id, node_id) else {
+            return Err(FederationError::internal(format!(
+                "Cannot merge {} into {}: the former is not a child of the latter",
+                child_id.index(),
+                node_id.index()
+            )));
+        };
+        let Some(child_path_in_this) = relation_to_child.path_in_parent else {
+            return Err(FederationError::internal(format!(
+                "Cannot merge {} into {}: the path of the former into the later is unknown",
+                child_id.index(),
+                node_id.index()
+            )));
+        };
+        self.merge_in_internal(node_id, child_id, &child_path_in_this, false)
+    }
+
+    /// Merges a grand child of `this` group into it.
+    ///
+    /// Note that it is up to the caller to know that doing such merging is reasonable in the first place, which
+    /// generally means knowing that 1) `grandChild.inputs` are included in `this.inputs` and 2) all of `grandChild.selection`
+    /// can safely be queried on the `this.subgraphName` subgraph (the later of which is trivially true if `this` and
+    /// `grandChild` are on the same subgraph and same mergeAt).
+    ///
+    /// @param grandChild - a group that must be a "grand child" (a child of a child) of `this`, and for which the
+    ///   'path in parent' is know for _both_ the grand child to tis parent and that parent to `this`. The `canMergeGrandChildIn`
+    ///     method can be used to ensure that `grandChild` meets those requirement.
+    fn merge_grand_child_in(
+        &mut self,
+        node_id: NodeIndex,
+        grand_child_id: NodeIndex,
+    ) -> Result<(), FederationError> {
+        let grand_child_parents: Vec<ParentRelation> =
+            self.parents_relations_of(grand_child_id).collect();
+        if grand_child_parents.len() != 1 {
+            return Err(FederationError::internal(format!(
+                "Cannot merge {} as it has multiple parents []",
+                grand_child_id.index()
+            )));
```

Comment:
Yeah. It's just a placeholder for now as just fetch nodes don't have display implementation atm so printing out just the IDs is pretty much useless info anyway.

Reply:
ok, gotcha. if we don't implement `Display` as part of this PR, can you make an issue in jira with `fed-next` label and give it a parent of [`FED-192`](https://apollographql.atlassian.net/browse/FED-192)

Reply:
FetchDependencyNodes can be displayed but you have to call it explicitly: `node.display(node_index)`. This is because the node index is key information about a node.